dace/libraries/standard/nodes/transpose.py

```diff
@@ -140,27 +146,37 @@ class ExpandTransposeOpenBLAS(ExpandTransformation):
     @staticmethod
     def expansion(node, state, sdfg):
         node.validate(sdfg, state)
+
+        # Fall back to native implementation if input and output types are not the same
+        if (sdfg.arrays[list(state.in_edges_by_connector(node, '_inp'))[0].data.data].dtype != sdfg.arrays[list(
+                state.out_edges_by_connector(node, '_out'))[0].data.data].dtype):
+            return ExpandTransposePure.make_sdfg(node, state, sdfg)
+
         dtype = node.dtype
         if dtype == dace.float32:
             func = "somatcopy"
             alpha = "1.0f"
+            cast = ''
         elif dtype == dace.float64:
             func = "domatcopy"
             alpha = "1.0"
+            cast = ''
         elif dtype == dace.complex64:
             func = "comatcopy"
             alpha = "dace::blas::BlasConstants::Get().Complex64Pone()"
+            cast = '(float*)'
         elif dtype == dace.complex128:
             func = "zomatcopy"
             alpha = "dace::blas::BlasConstants::Get().Complex128Pone()"
+            cast = '(double*)'
```

Comment:
I have the impression that this has been fixed in another PR, but maybe it is stuck in the queue. Just noting it in case merge issues appear later.

Reply:
It was fixed for some BLAS operators but not others. Also there is some tensor transpose PR but I think it’s GPU focused.